cmd/aws-s3-csi-mounter/csimounter/csimounter.go

```diff
@@ -37,11 +39,16 @@ func Run(options Options) (int, error) {
 	mountOptions := options.MountOptions
 	mountpointArgs := mountpoint.ParseArgs(mountOptions.Args)
 
-	// TODO: This is a temporary workaround to create a cache folder if caching is enabled,
-	// ideally we should create a volume (`emptyDir` by default) in the Mountpoint Pod and use that.
-	mountpointArgs, err := createCacheDir(mountpointArgs)
-	if err != nil {
-		return 0, fmt.Errorf("failed to create cache dir: %w", err)
+	localCacheDir := filepath.Join("/", mppod.LocalCacheDirName)
+	_, localCacheEnabledViaMountOptions := mountpointArgs.Remove(mountpoint.ArgCache)
+	localCacheMounted := checkIfDirExists(localCacheDir)
+
+	if localCacheEnabledViaMountOptions && !localCacheMounted {
+		return 0, fmt.Errorf("local cache enabled via mount options but cache folder is not mounted at %q", localCacheDir)
+	}
+
+	if localCacheMounted {
+		mountpointArgs.Set(mountpoint.ArgCache, localCacheDir)
```

Comment:
np: Not sure I like to modify `mountpointArgs` in `csimounter` in addition to modifying them in `node`. Is there a reason we can't adjust `--cache` argument on the Node side?

Reply:
I think we don't have this information in the node component, maybe we can move that logic from the node component to here as well?

Reply:
By "this information" you mean `--cache /local-cache` flag? Node should have this in mountOptions from PV, no?

Reply:
It's the fact that `local-cache` volume is mounted to Mountpoint Pod. We can also understand by looking into mount options and volume attributes to understand if caching is enabled, but that would mean we need to duplicate the same logic in both controller and node. That's why it feels easier to make that decision on Mountpoint Pod by looking whether the cache volume is mounted at `/local-cache` or not.